ibrcommon/ibrcommon/thread/RWMutex.cpp

```diff
@@ -21,17 +21,58 @@
 
 #include "ibrcommon/thread/RWMutex.h"
 #include <errno.h>
+#include <cstring>
 
 namespace ibrcommon
 {
 	RWMutex::RWMutex()
 	{
-		pthread_rwlock_init(&_rwlock, NULL);
+		std::memset(&_rwlock,0,sizeof(pthread_rwlock_t));
```

Comment:
memset belongs to the default namespace. Hence, the right call would be ::memset(...).